### DISCUSSION TITLE
Share a App Store Connect API key with multiple developers?

### DISCUSSION THREAD

**Discussion by oonoo** [fastlane/fastlane#18093] [Q&A]
Is there already something like an equivalent of fastlane match for sharing a App Store Connect API key available?

How do other developers share the API key? 
The current alternative that comes to my mind is managing for every developer an Apple ID with an API key. But it feels like a lot of effort for a big company.

**Comment by dinsen** [comment]
Hi @oonoo 
Currently there is nothing to handle sharing of API Keys.

I'm also curious how other companies will handle multiple keys for different projects.
For now in my company we have made a way of storing keys in a private repo like match do with certificates.

**Reply by oonoo** [reply]
Would be nice to have something like match for API key sharing: handling an encrypted repo and copying API key file to the "correct"/configured place, similar to what match does. The encryption would make it safer than just a repo, and the copying mechanism of the file would make sure it is in the same place on every developer machine.

For now I just implemented an action called `load_json_to_app_store_connect_api_key` which reads a API key JSON file (from a hard coded path) and passes it to the `app_store_connect_api_key` action. `app_store_connect_api_key` sets `SharedValues::APP_STORE_CONNECT_API_KEY` which is read by deliver, match and so on to create the token. That works fine on the CI where I just place the API key JSON file to the path where `load_json_to_app_store_connect_api_key` expects it. But now I am wondering how we can enable other developer to run fastlanes with apple server access.

I feel like "match for API keys" could be a feature more developers could use. I mean you created your solution with the private repo and I will maybe also create some solution for this.
